benchalerts/benchalerts/parse_conbench.py

```diff
@@ -182,7 +182,13 @@ def regression_details(comparisons: List[RunComparison]) -> str:
     if not any(comparison.baseline_info for comparison in comparisons):
         return None
 
-    z_score_threshold = comparisons[0].compare_results[0]["threshold_z"]
+    # Find the z-score threshold that was used so we can display it here
+    z_score_threshold = None
+    for comparison in comparisons:
+        if comparison.compare_results:
+            z_score_threshold = comparison.compare_results[0]["z_score_threshold"]
```

Comment:
These should all be the same, yeah? would something like this (+ possibly a check that the set is a single element) make sense + be easier to debug?

```
z_score_threshold  = set([x.compare_results[0]["threshold_z"] for x in comparisons if x is not None])
```

Reply:
Yes, they should be the same. This should be guaranteed, but there's not one nice code line I can point to to show that guarantee, so I'll do your `set()` suggestion so that it's obvious!